### PR TITLE
ConvLSTM2D extend Recurrent

### PR DESCRIPTION
**Pull request by hkvision**
enhancement for #724 

ConvLSTM2D should extend Recurrent instead of KerasLayer.

### CODE REVIEW

**Comment by hkvision** [comment]
jenkins passed http://10.239.47.211:18888/view/ZOO-PR/job/ZOO-PR-Validation/1281/

File: zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras/layers/ConvLSTM2D.scala [hunk 1]
```diff
@@ -54,28 +56,55 @@ import scala.reflect.ClassTag
...
+   val subsample: Int = 1,
+   var wRegularizer: Regularizer[T] = null,
+   var uRegularizer: Regularizer[T] = null,
+   var bRegularizer: Regularizer[T] = null,
+   var returnSeq: Boolean = false,
+   var goBackward: Boolean = false,
+   var mInputShape: Shape = null)(implicit ev: TensorNumeric[T])
+  extends Recurrent[T](outputDimension, returnSeq, goBackward, mInputShape) with Net {
+
+  require(dimOrdering.toLowerCase() == "channel_first", s"ConvLSTM2D currently only supports " +
```
Comment (dding3) [inline zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras/layers/ConvLSTM2D.scala]: If we only support "channel_first", maybe we should not expose `dimOrdering`, if we need have the same parameters with keras, we should support other dimOrdering. 

BTW, do we need keep the same parameter name with keras? Looks like they're different. https://keras.io/layers/recurrent/#convlstm2d

File: zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras/layers/ConvLSTM2D.scala [hunk 2]
```diff
@@ -54,28 +56,55 @@ import scala.reflect.ClassTag
...
-  extends BigDLConvLSTM2D[T](nbFilter, nbKernel, activation, innerActivation, dimOrdering,
-                              subsample, wRegularizer, uRegularizer, bRegularizer, returnSequences,
-                              goBackwards, inputShape) with Net {}
+   var outputDimension: Int,
+   val nbKernel: Int,
+   val activation: KerasLayer[Tensor[T], Tensor[T], T] = null,
+   val innerActivation: KerasLayer[Tensor[T], Tensor[T], T] = null,
+   val dimOrdering: String = "CHANNEL_FIRST",
+   val subsample: Int = 1,
+   var wRegularizer: Regularizer[T] = null,
```
Comment (dding3) [inline zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras/layers/ConvLSTM2D.scala]: miss padding?

File: zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras/layers/ConvLSTM2D.scala (hunk 1 above)
Comment (hkvision) [reply zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras/layers/ConvLSTM2D.scala]: What you are looking is Keras 2.x, here we refer to Keras 1.2.2

File: zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras/layers/ConvLSTM2D.scala (hunk 2 above)
Comment (hkvision) [reply zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras/layers/ConvLSTM2D.scala]: In Keras, it uses `border_mode` = 'same'/'valid' for padding. Seems we only support 'same' and I ignore this argument. Alternatively, we can add a border_mode argument with only one available value like dimOrdering lol if you wish...

**Comment by hkvision** [comment]
Let's continue on this PR cc @dding3

File: zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras/layers/ConvLSTM2D.scala (hunk 2 above)
Comment (dding3) [reply zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras/layers/ConvLSTM2D.scala]: Let's keep current implementation. After support `valid` padding, we will support user config padding way.

**Comment by dding3** [comment]
Generally looks good to me. Please stress we only support "same" padding currently.

**Comment by hkvision** [comment]
Highlighted only same padding is currently supported. We can add "borderMode" arg after support same padding. @dding3

**Comment by hkvision** [comment]
jenkins passed http://10.239.47.211:18888/view/ZOO-PR/job/ZOO-PR-Validation/1848/
Thanks for your review @dding3
